dbuf/dbuf/client.py

```diff
@@ -0,0 +1,169 @@
+import random
+import aiohttp
+import collections
+import struct
+
+import hailtop.utils as utils
+from hailtop.config import get_deploy_config
+
+
+class DBufClient:
+    async def __aenter__(self):
+        self.aiosession = aiohttp.ClientSession(raise_for_status=True,
+                                                timeout=aiohttp.ClientTimeout(total=60))
+        return self
+
+    async def __aexit__(self, exc_type, exc, tb):
+        await self.aiosession.close()
+
+    def __init__(self,
+                 leader_name,
+                 id=None,
+                 max_bufsize=10 * 1024 * 1024 - 1,
+                 deploy_config=None,
+                 rng=None):
+        if not deploy_config:
+            deploy_config = get_deploy_config()
+        if rng is None:
+            rng = random.Random()
+        self.rng = rng
+        self.deploy_config = deploy_config
+        self.root_url = deploy_config.base_url(leader_name)
+        self.session_url = None if id is None else f'{self.root_url}/s/{id}'
+        self.aiosession = None
+        self.id = id
+        self.offs = []
+        self.sizes = []
+        self.cursor = 0
+        self.max_bufsize = max_bufsize
+
+    async def create(self):
+        resp = await utils.request_retry_transient_errors(
+            self.aiosession,
+            'POST',
+            f'{self.root_url}/s')
+        assert resp.status == 200
+        self.id = int(await resp.text())
+        self.session_url = f'{self.root_url}/s/{self.id}'
+        return self.id
+
+    async def start_write(self):
+        workers = await self.get_workers()
+        worker = workers[self.rng.randrange(len(workers))]
+        return DBufAppender(worker, self.deploy_config, self.id, self.max_bufsize, self.aiosession)
+
+    async def get(self, key):
+        server = key[0]
+        server_url = self.deploy_config.base_url(server)
+
+        resp = await utils.request_retry_transient_errors(
+            self.aiosession,
+            'POST',
+            f'{server_url}/s/{self.id}/get',
+            json=key)
+        assert resp.status == 200
```

Comment:
As Cotton said, this is still unnecessary in presence of `raise_for_status = True` above.